datadog/resource_datadog_synthetics_test_.go

```diff
@@ -181,6 +181,24 @@ func syntheticsTestRequest() *schema.Resource {
 				Type:        schema.TypeBool,
 				Optional:    true,
 			},
+			"number_of_packets": {
+				Description: "Number of pings to use per test for ICMP tests (`subtype = \"icmp\"`).",
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
```

Comment:
Let's directly use https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/helper/validation#IntBetween
And also let's add the allowed range to the description